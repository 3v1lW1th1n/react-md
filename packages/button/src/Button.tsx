import * as React from "react";
import * as PropTypes from "prop-types";
import cn from "classnames";
import { StatesConsumer } from "@react-md/states";
import { TextIconSpacing, ITextIconSpacingProps } from "@react-md/icon";
import { KeyboardClickable } from "@react-md/a11y";

export interface IButtonThemeProps {
  /**
   * An optional className to apply.
   *
   * @docgen
   */
  className?: string;

  /**
   * Boolean if the button is currently disabled.
   *
   * @docgen
   */
  disabled?: boolean;

  /**
   * The specific material design button type to use. A text button will display
   * text while an icon button will only contain an icon.
   *
   * @docgen
   */
  btnType?: "text" | "icon";

  /**
   * The material design theme to apply to the button.
   *
   * @docgen
   */
  theme?: "clear" | "primary" | "secondary" | "default";

  /**
   * The material design theme type to apply.
   *
   * @docgen
   */
  themeType?: "flat" | "outline" | "contained";
}

/**
 * The `Button` component is used to create a clickable area within your application. It can be styled
 * to be flat with the background, outlined, or contained. A contained button will include some elevation
 * to help increase its visibility within the app. In addition, the button can be themed to either be clear,
 * a default grey color, or to use the app's defined primary or secondary color.
 *
 * Buttons come in the form of text, icon, or text and icon together. It is recommended to use the text
 * version when possible since it is less confusing for the user, but icons can be used if they are easy
 * to understand and there is limited space.
 *
 * Another feature of the `Button` is that it can be conditionally rendered as a `<div>` instead of a `<button>`
 * if you need to create a more advanced clickable area that has `<div>`s inside (it is considered invalid html to have
 * a `<div>` within a `<button>`). This will make the div fully accessible to keyboard users and add the correct
 * keyboard events.
 */
export interface IButtonButtonProps
  extends IButtonThemeProps,
    ITextIconSpacingProps,
    React.HTMLAttributes<HTMLButtonElement | HTMLDivElement> {
  /**
   * An optional style to apply.
   *
   * @docgen
   */
  style?: React.CSSProperties;

  /**
   * The button's type attribute.
   *
   * @docgen
   */
  type?: "button" | "reset" | "submit";

  /**
   * Any children to render within the button. This will normally just be text or an icon.
   *
   * Please note that it is considered invalid html to have a `<div>` as a descendant of a `<button>`.
   * You can fix this by enabling the `asDiv` prop.
   *
   * @docgen
   */
  children?: React.ReactNode;

  /**
   * Boolean if the button should be rendered as a div instead. This will update the div to be fully
   * accessible with the [button role](https://www.w3.org/TR/wai-aria-practices/#button). If you want
   * to have a `<div>` as a child of the button, you should enable this prop.
   *
   * @docgen
   */
  asDiv?: boolean;

  // these are really just pass-through-values, so don't need to document them about how they are passed
  // to the helper components
  onBlur?: (event: React.FocusEvent<HTMLButtonElement | HTMLDivElement>) => void;
  onMouseDown?: (event: React.MouseEvent<HTMLButtonElement | HTMLDivElement>) => void;
  onMouseUp?: (event: React.MouseEvent<HTMLButtonElement | HTMLDivElement>) => void;
  onTouchStart?: (event: React.TouchEvent<HTMLButtonElement | HTMLDivElement>) => void;
  onTouchEnd?: (event: React.TouchEvent<HTMLButtonElement | HTMLDivElement>) => void;
  onKeyDown?: (event: React.KeyboardEvent<HTMLButtonElement | HTMLDivElement>) => void;
  onKeyUp?: (event: React.KeyboardEvent<HTMLButtonElement | HTMLDivElement>) => void;
}

export type IButtonProps = IButtonButtonProps | IButtonButtonProps;

export interface IButtonDefaultProps {
  asDiv: boolean;
  disabled: boolean;
  theme: string;
  themeType: string;
  btnType: string;
  iconAfter: boolean;
}

export type ButtonWithDefaultProps = IButtonProps & IButtonDefaultProps;

export default class Button extends React.Component<IButtonProps, {}> {
  public static propTypes = {
    style: PropTypes.object,
    className: PropTypes.string,
    type: PropTypes.oneOf(["button", "reset", "submit"]),
    btnType: PropTypes.oneOf(["text", "icon"]),
    theme: PropTypes.oneOf(["clear", "primary", "secondary", "default"]),
    themeType: PropTypes.oneOf(["flat", "outline", "contained"]),
    children: PropTypes.node,
    icon: PropTypes.oneOfType([PropTypes.element, PropTypes.node]),
    iconAfter: PropTypes.bool,
  };

  public static defaultProps: IButtonDefaultProps = {
    asDiv: false,
    disabled: false,
    theme: "primary",
    themeType: "flat",
    btnType: "text",
    iconAfter: false,
  };

  /**
   * Creates a button theme based on the button theming props. This is really just used so that
   * other elements like clickable `<div>`s or `<input type="file">` can look like buttons.
   *
   * NOTE: You will still need to manually apply the `<StatesConsumer>` with the correct props
   */
  public static theme(props: IButtonThemeProps): string {
    const { btnType, themeType, theme, disabled, className } = props;
    const text = btnType === "text";
    const icon = btnType === "icon";
    const flat = themeType === "flat";
    const outline = themeType === "outline";
    const contained = themeType === "contained";
    const primary = theme === "primary";
    const secondary = theme === "secondary";
    const defaultTheme = theme === "default";

    return cn(
      "rmd-btn",
      {
        "rmd-btn--text": text,
        "rmd-btn--icon": icon,
        "rmd-btn--hoverable": !disabled,
        "rmd-btn--disabled": disabled,
        "rmd-btn--primary": !disabled && primary && contained,
        "rmd-btn--secondary": !disabled && secondary && contained,
        "rmd-btn--default": !disabled && defaultTheme && contained,
        "rmd-btn--text-primary": !disabled && primary && !contained,
        "rmd-btn--text-secondary": !disabled && secondary && !contained,
        "rmd-btn--text-default": !disabled && defaultTheme && !contained,
        "rmd-btn--outline-primary": !disabled && primary && outline,
        "rmd-btn--outline-secondary": !disabled && secondary && outline,
        "rmd-btn--outline-default": !disabled && defaultTheme && outline,
        "rmd-btn--contained": !disabled && contained,
      },
      className
    );
  }

  public render() {
    const {
      theme,
      themeType,
      btnType,
      icon,
      iconAfter,
      children,
      disabled: baseDisabled,
      onMouseDown,
      onMouseUp,
      onTouchStart,
      onTouchEnd,
      onKeyDown,
      onKeyUp,
      asDiv,
      ...props
    } = this.props;

    return (
      <StatesConsumer
        className={Button.theme(this.props)}
        pressedClassName={cn({ "rmd-btn--contained-pressed": themeType === "contained" })}
        disabled={baseDisabled}
        onMouseDown={onMouseDown}
        onMouseUp={onMouseUp}
        onTouchStart={onTouchStart}
        onTouchEnd={onTouchEnd}
        onKeyDown={onKeyDown}
        onKeyUp={onKeyUp}
      >
        {({ disabled, ...stateProps }) => {
          // removed `disabled` since it is "invalid" to have a disabled attribute on a div even if it
          // can be rendered. Instead we will apply "aria-disabled" when it is disabled from KeyboardClickable

          const content = (
            <TextIconSpacing icon={icon} iconAfter={iconAfter}>
              {children}
            </TextIconSpacing>
          );

          if (asDiv) {
            return (
              <KeyboardClickable disabled={disabled} {...stateProps}>
                {clickableProps => (
                  <div {...props} {...stateProps} {...clickableProps}>
                    {content}
                  </div>
                )}
              </KeyboardClickable>
            );
          }

          return (
            <button {...props} {...stateProps} disabled={disabled}>
              {content}
            </button>
          );
        }}
      </StatesConsumer>
    );
  }
}
