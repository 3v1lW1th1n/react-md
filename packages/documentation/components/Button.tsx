import React, { FunctionComponent } from "react";
import {
  Button as RMDButton,
  ButtonProps as RMDButtonProps,
} from "@react-md/button";
import { Tooltipped, TooltippedProps } from "@react-md/tooltip";
import { Omit } from "@react-md/utils";

export interface ButtonProps
  extends Omit<RMDButtonProps, "id">,
    Pick<TooltippedProps, "id" | "tooltip"> {}

const Button: FunctionComponent<ButtonProps> = ({
  id,
  tooltip: propTooltip,
  onClick,
  onMouseEnter,
  onMouseLeave,
  children,
  ...props
}) => (
  <Tooltipped
    id={id}
    tooltip={propTooltip}
    onClick={onClick}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
  >
    {({ tooltip, containerProps }) => (
      <RMDButton {...props} {...containerProps}>
        {children}
        {tooltip}
      </RMDButton>
    )}
  </Tooltipped>
);

export default Button;
