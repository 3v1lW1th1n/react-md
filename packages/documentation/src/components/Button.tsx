import React, { FC, ReactNode } from "react";
import { Button as RMDButton, ButtonProps } from "@react-md/button";
import { Tooltipped } from "@react-md/tooltip";

interface Props extends ButtonProps {
  id: string;
  tooltip?: ReactNode;
}

const Button: FC<Props> = ({
  id,
  tooltip,
  onMouseEnter,
  onMouseLeave,
  onTouchStart,
  onTouchMove,
  onFocus,
  onKeyDown,
  children,
  ...props
}) => (
  <Tooltipped
    id={id}
    tooltip={tooltip}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
    onTouchStart={onTouchStart}
    onTouchMove={onTouchMove}
    onFocus={onFocus}
    onKeyDown={onKeyDown}
  >
    <RMDButton {...props}>{children}</RMDButton>
  </Tooltipped>
);

export default Button;
