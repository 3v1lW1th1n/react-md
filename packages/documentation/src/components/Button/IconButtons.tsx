import * as React from "react";
import { Button } from "@react-md/button";
import { FontIcon } from "@react-md/icon";
import { FavoriteSVGIcon, AspectRatioFontIcon } from "@react-md/material-icons";
import { Text } from "@react-md/typography";

const IconButtons: React.SFC<any> = () => (
  <React.Fragment>
    <Text type="headline-6">Theme Examples</Text>
    <Button id="icon-button-1" className="example-group__example" btnType="icon">
      <FavoriteSVGIcon />
    </Button>
    <Button id="icon-button-2" className="example-group__example" btnType="icon" theme="primary">
      <AspectRatioFontIcon />
    </Button>
    <Button id="icon-button-3" className="example-group__example" btnType="icon" theme="secondary">
      <FontIcon iconClassName="fa fa-star-o" />
    </Button>
    <Button id="icon-button-4" className="example-group__example" btnType="icon" themeType="outline">
      <FavoriteSVGIcon />
    </Button>
    <Button id="icon-button-5" className="example-group__example" btnType="icon" themeType="outline" theme="primary">
      <AspectRatioFontIcon />
    </Button>
    <Button id="icon-button-6" className="example-group__example" btnType="icon" themeType="outline" theme="secondary">
      <FontIcon iconClassName="fa fa-star-o" />
    </Button>
    <Button id="icon-button-7" className="example-group__example" btnType="icon" themeType="contained">
      <FavoriteSVGIcon />
    </Button>
    <Button id="icon-button-8" className="example-group__example" btnType="icon" themeType="contained" theme="primary">
      <AspectRatioFontIcon />
    </Button>
    <Button
      id="icon-button-9"
      className="example-group__example"
      btnType="icon"
      themeType="contained"
      theme="secondary"
    >
      <FontIcon iconClassName="fa fa-star-o" />
    </Button>
    <Text type="headline-6">Disabled Examples</Text>
    <Button id="icon-button-10" className="example-group__example" btnType="icon" disabled={true}>
      <FavoriteSVGIcon />
    </Button>
    <Button id="icon-button-11" className="example-group__example" btnType="icon" disabled={true} theme="primary">
      <AspectRatioFontIcon />
    </Button>
    <Button id="icon-button-12" className="example-group__example" btnType="icon" disabled={true} theme="secondary">
      <FontIcon iconClassName="fa fa-star-o" />
    </Button>
    <Button id="icon-button-13" className="example-group__example" btnType="icon" disabled={true} themeType="outline">
      <FavoriteSVGIcon />
    </Button>
    <Button
      id="icon-button-14"
      className="example-group__example"
      btnType="icon"
      disabled={true}
      themeType="outline"
      theme="primary"
    >
      <AspectRatioFontIcon />
    </Button>
    <Button
      id="icon-button-15"
      className="example-group__example"
      btnType="icon"
      disabled={true}
      themeType="outline"
      theme="secondary"
    >
      <FontIcon iconClassName="fa fa-star-o" />
    </Button>
    <Button id="icon-button-16" className="example-group__example" btnType="icon" disabled={true} themeType="contained">
      <FavoriteSVGIcon />
    </Button>
    <Button
      id="icon-button-17"
      className="example-group__example"
      btnType="icon"
      disabled={true}
      themeType="contained"
      theme="primary"
    >
      <AspectRatioFontIcon />
    </Button>
    <Button
      id="icon-button-18"
      className="example-group__example"
      btnType="icon"
      disabled={true}
      themeType="contained"
      theme="secondary"
    >
      <FontIcon iconClassName="fa fa-star-o" />
    </Button>
    <Text type="headline-6">Rendering as Divs Examples</Text>
    <Button id="icon-button-19" className="example-group__example" btnType="icon" asDiv={true}>
      <FavoriteSVGIcon />
    </Button>
    <Button id="icon-button-20" className="example-group__example" btnType="icon" asDiv={true} theme="primary">
      <AspectRatioFontIcon />
    </Button>
    <Text type="headline-6">Rendering as disabled Divs Examples</Text>
    <Button id="icon-button-21" className="example-group__example" btnType="icon" asDiv={true} disabled={true}>
      <FavoriteSVGIcon />
    </Button>
    <Button
      id="icon-button-22"
      className="example-group__example"
      btnType="icon"
      asDiv={true}
      theme="primary"
      disabled={true}
    >
      <AspectRatioFontIcon />
    </Button>
  </React.Fragment>
);

export default IconButtons;
