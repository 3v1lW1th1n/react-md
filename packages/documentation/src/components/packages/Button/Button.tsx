import * as React from "react";
import * as Router from "react-router";
import { Switch, Route, Redirect } from "react-router-dom";

import Loadable from "components/Loadable";
import NotFound from "components/NotFound";

const Examples = Loadable(() => import("./Examples"));
const ButtonPropTypes = Loadable(() => import("./ButtonPropTypes"));
const ButtonSassDoc = Loadable(() => import("./ButtonSassDoc"));

const Button: React.SFC<Router.RouteComponentProps<void>> = ({ match }) => (
  <Switch>
    <Redirect exact={true} from={match.path} to={`${match.path}/installation`} />
    <Route path={`${match.path}/examples`} component={Examples} />
    <Route path={`${match.path}/proptypes`} component={ButtonPropTypes} />
    <Route path={`${match.path}/sassdoc`} component={ButtonSassDoc} />
    <Route component={NotFound} />
  </Switch>
);

export default Button;
