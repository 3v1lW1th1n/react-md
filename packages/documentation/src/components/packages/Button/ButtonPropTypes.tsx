import * as React from "react";

import PropTypesPage from "components/PropTypesPage";

// const props = {};

const ButtonPropTypes = () => <PropTypesPage name="Button" />;

export default ButtonPropTypes;
