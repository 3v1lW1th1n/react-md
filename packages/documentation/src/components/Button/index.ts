import { default as Buttons } from "./Button";

export default Buttons;
