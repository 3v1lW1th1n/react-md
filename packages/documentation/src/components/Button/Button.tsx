import * as React from "react";

import ExamplesPage from "../ExamplesPage";

import FlatButtons from "./FlatButtons";
import OutlinedButtons from "./OutlinedButtons";
import ContainedButtons from "./ContainedButtons";
import IconButtons from "./IconButtons";

const examples = [{
  title: "Flat Buttons",
  children: <FlatButtons />,
}, {
  title: "Outlined Buttons",
  children: <OutlinedButtons />,
}, {
  title: "Contained Buttons",
  children: <ContainedButtons />,
}, {
  title: "Icon Buttons",
  children: <IconButtons />,
}];

const Buttons: React.SFC<{}> = () => <ExamplesPage examples={examples} title="Buttons" />;

export default Buttons;
