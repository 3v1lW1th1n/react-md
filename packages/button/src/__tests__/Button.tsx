import * as React from "react";
import { create } from "react-test-renderer";

import Button from "../Button";
import {
  ButtonTheme,
  ButtonThemeType,
  ButtonType,
  IButtonThemeProps,
} from "../types";

const themes: ButtonTheme[] = [
  "clear",
  "primary",
  "secondary",
  "warning",
  "error",
];
const themeTypes: ButtonThemeType[] = ["flat", "contained", "outline"];
const buttonTypes: ButtonType[] = ["text", "icon"];

function flattenDeep(
  arr: any[]
): (IButtonThemeProps & { children: React.ReactNode })[] {
  return arr.reduce(
    (flattened, val) =>
      Array.isArray(val)
        ? flattened.concat(flattenDeep(val))
        : flattened.concat(val),
    []
  );
}

describe("Button", () => {
  // this is actually really bad practice and kind of worthess
  it("should render correctly based on the theme props", () => {
    flattenDeep(
      buttonTypes.map(buttonType =>
        themes.map(theme =>
          themeTypes.map(themeType => ({
            buttonType,
            theme,
            themeType,
            children: buttonType === "text" ? "Button Text" : <i />,
          }))
        )
      )
    ).forEach(themeProps => {
      expect(create(<Button {...themeProps} />).toJSON()).toMatchSnapshot();
    });
  });
});
