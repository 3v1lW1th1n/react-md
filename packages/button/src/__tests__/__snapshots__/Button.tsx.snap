// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly based on the theme props 1`] = `
<button
  class="rmd-button rmd-button--text"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 2`] = `
<button
  class="rmd-button rmd-button--text rmd-button--contained"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 3`] = `
<button
  class="rmd-button rmd-button--text rmd-button--outline"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 4`] = `
<button
  class="rmd-button rmd-button--text rmd-button--text-primary"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 5`] = `
<button
  class="rmd-button rmd-button--text rmd-button--contained rmd-button--primary"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 6`] = `
<button
  class="rmd-button rmd-button--text rmd-button--outline rmd-button--text-primary rmd-button--outline-primary"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 7`] = `
<button
  class="rmd-button rmd-button--text rmd-button--text-secondary"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 8`] = `
<button
  class="rmd-button rmd-button--text rmd-button--contained rmd-button--secondary"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 9`] = `
<button
  class="rmd-button rmd-button--text rmd-button--outline rmd-button--text-secondary rmd-button--outline-secondary"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 10`] = `
<button
  class="rmd-button rmd-button--text rmd-button--text-warning"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 11`] = `
<button
  class="rmd-button rmd-button--text rmd-button--contained rmd-button--warning"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 12`] = `
<button
  class="rmd-button rmd-button--text rmd-button--outline rmd-button--text-warning rmd-button--outline-warning"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 13`] = `
<button
  class="rmd-button rmd-button--text rmd-button--text-error"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 14`] = `
<button
  class="rmd-button rmd-button--text rmd-button--contained rmd-button--error"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 15`] = `
<button
  class="rmd-button rmd-button--text rmd-button--outline rmd-button--text-error rmd-button--outline-error"
>
  Button Text
</button>
`;

exports[`Button should render correctly based on the theme props 16`] = `
<button
  class="rmd-button rmd-button--icon"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 17`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--contained"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 18`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--outline"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 19`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--text-primary"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 20`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--contained rmd-button--primary"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 21`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--outline rmd-button--text-primary rmd-button--outline-primary"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 22`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--text-secondary"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 23`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--contained rmd-button--secondary"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 24`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--outline rmd-button--text-secondary rmd-button--outline-secondary"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 25`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--text-warning"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 26`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--contained rmd-button--warning"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 27`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--outline rmd-button--text-warning rmd-button--outline-warning"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 28`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--text-error"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 29`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--contained rmd-button--error"
>
  <i />
</button>
`;

exports[`Button should render correctly based on the theme props 30`] = `
<button
  class="rmd-button rmd-button--icon rmd-button--outline rmd-button--text-error rmd-button--outline-error"
>
  <i />
</button>
`;
